[all hosts] Add Glossary of terms article (#3161)
OfficeDev/office-js-docs-pr: docs/resources/resources-glossary.md

---
title: Office Add-ins glossary of terms
description: A glossary of terms commonly used throughout the Office Add-ins documentation.
ms.date: 02/04/2022
ms.localizationpriority: medium
---

# Office Add-ins glossary of terms

This is a glossary of terms commonly used throughout the Office Add-ins documentation.

> ## add-in

Author: I'll let our editor make the final call on this. I don't see any guidance in our style guide for how to format a glossary, but our style guide itself doesn't capitalize nouns in this context unless they're proper nouns. Term Studio also only capitalizes proper nouns. 

I think following the style of a typical dictionary makes the most sense and teaches our customers how to use the terms properly in context by distinguishing proper nouns from nouns.

> ## add-in

Author: @lindalu-MSFT Can you take a look at this conversation and provide feedback from an editorial perspective?

> ## add-in

Reviewer: Initially, I thought I agreed with Alex that the terms should be capitalized in the headings. But then I located several glossaries that follow this guidance, so now I agree more with Alison.: "The glossary uses standard (dictionary-style) capitalization in the terms list: each glossary term should be lowercase unless it is treated as a proper noun, in which case it should be capitalized."

Office Add-ins are web applications that extend Office applications. These web applications add new functionality to the Office application, such as bring in external data, automate processes, or embed interactive objects in Office documents.

Office Add-ins differ from VBA, COM, and VSTO add-ins because they offer cross-platform support (web, Windows, Mac, and iPad) and are based on standard web technologies (HTML, CSS, and JavaScript). The primary programming language of an Office Add-in is JavaScript or TypeScript.

## add-in commands

> **Add-in commands** are UI elements, such as buttons and menus, that extend the Office UI for your add-in. When users select an add-in command element, they initiate actions such as running JavaScript code or displaying the add-in in a task pane. Add-in commands let your add-in look and feel like a part of Office, which gives users more confidence in your add-in. See [Add-in commands for Excel, PowerPoint, and Word](../design/add-in-commands.md) to learn more.

Reviewer 2: Maybe we can also include a link to the Outlook article...
```suggestion
**Add-in commands** are UI elements, such as buttons and menus, that extend the Office UI for your add-in. When users select an add-in command element, they initiate actions such as running JavaScript code or displaying the add-in in a task pane. Add-in commands let your add-in look and feel like a part of Office, which gives users more confidence in your add-in. See [Add-in commands for Excel, PowerPoint, and Word](../design/add-in-commands.md) and [Add-in commands for Outlook](../outlook/add-in-commands-for-outlook.md) to learn more.
```

See also: [ribbon, ribbon button](#ribbon-ribbon-button).

## application

In the Office Add-ins documentation, **application** refers to an Office application. The Office applications that support Office Add-ins are: Excel, OneNote, Outlook, PowerPoint, Project, Visio, and Word.

See also: [client](#client), [host](#host), [Office application, Office client](#office-application-office-client).

## application-specific API

Application-specific APIs provide strongly-typed objects that can be used to interact with objects that are native to a specific Office application. For example, you can use the Excel JavaScript APIs to access worksheets, ranges, tables, charts, and more. Application-specific APIs are currently available for Excel, OneNote, PowerPoint, and Word. See [Application-specific API model](../develop/application-specific-api-model.md) to learn more.

See also: [Common API](#common-api).

## CDN

CDN is an acronym. It represents **content delivery network (CDN)** and refers to a distributed network of servers and data centers. A CDN typically provides higher resource availability and performance when compared to a single server or data center.

## client

> In the Office Add-ins documentation, **client** typically refers to an Office application. The Office applications, or clients, that support Office Add-ins are: Excel, OneNote, Outlook, PowerPoint, Project, Visio, and Word.

Author: I don't understand, can you elaborate? Does **client** differ from **application** and **host** in this way?

> In the Office Add-ins documentation, **client** typically refers to an Office application. The Office applications, or clients, that support Office Add-ins are: Excel, OneNote, Outlook, PowerPoint, Project, Visio, and Word.

Reviewer 3: There's a distinction between **client** and **server** in most web services documentation. There isn't a distinction in our docs between Office clients, applications, and hosts, but there is a distinction between the end-user's Office client and the web service hosting the add-in. What you have is true; I was just looking for a place to inject more context about the add-ins model.

Feel free to ignore this comment if you feel **client** is usually used as a synonym for Office application in our docs.

See also: [application](#application), [host](#host), [Office application, Office client](#office-application-office-client).

## Common API

> Common APIs can be used to access features such as UI, dialogs, and client settings that are common across multiple types of Office applications. This API model uses [callbacks](https://developer.mozilla.org/docs/Glossary/Callback_function), which allow you to specify only one operation in each request sent to the Office application. Common APIs were introduced with Office 2013 and can be used to interact with Office 2013 or later. For details about the Common API object model, which includes APIs for interacting with Outlook, PowerPoint, and Project, see [Common JavaScript API object model](../develop/office-javascript-api-object-model.md).

Author: I don't feel strongly one way or the other about this, but I did pull the definition from the !IMPORTANT note at the top of this page: https://docs.microsoft.com/en-us/office/dev/add-ins/develop/office-javascript-api-object-model

See also: [application-specific API](#application-specific-api).

## Contoso

> **Contoso** Ltd. (also known as Contoso and Contoso University) is a fictional company used by Microsoft as an example company and domain.

Reviewer 2: I disagree about linking to this. It seems like an example M365 enterprise put together for their purpose.

## content add-in

> **Content add-ins** are surfaces that can be embedded directly into Excel or PowerPoint documents. Content add-ins give users access to interface controls that run code to modify documents or display data from a data source. Use content add-ins when you want to embed functionality directly into the document. See [Content Office Add-ins](../design/content-add-ins.md) to learn more.

Reviewer 4: I find the term "surfaces" mysterious. These are embedded webviews aren't they?

Author: I'm not familiar with the term "webview", and I don't see internet search results that consistently define this term. In term studio, I see the definition of webview is "A view that displays web content inside your application." Is this a term broadly understood by developers? I'm hesitant to use it, but I could use "webview" in this definition and then also define webview in the glossary. 

For reference, I pulled this definition: from https://docs.microsoft.com/en-us/office/dev/add-ins/design/content-add-ins

Reviewer 4: > I'm not familiar with the term "webview", and I don't see internet search results that consistently define this term. In term studio, I see the definition of webview is "A view that displays web content inside your application." Is this a term broadly understood by developers? I'm hesitant to use it, but I could use "webview" in this definition and then also define webview in the glossary.
> 
> I agree that "surfaces" is ambiguous.
> 
> For reference, I pulled this definition: from https://docs.microsoft.com/en-us/office/dev/add-ins/design/content-add-ins

We use "webview" elsewhere in our docs. Sometimes we say "web control" or "web view control" or "embedded browser". People are familiar with it. The task pane is a webview. We should change that article to use "webview" too.

> **Content add-ins** are surfaces that can be embedded directly into Excel or PowerPoint documents. Content add-ins give users access to interface controls that run code to modify documents or display data from a data source. Use content add-ins when you want to embed functionality directly into the document. See [Content Office Add-ins](../design/content-add-ins.md) to learn more.

Reviewer 2: I thought so too but the [Specify Office hosts...](https://docs.microsoft.com/office/dev/add-ins/develop/specify-office-hosts-and-api-requirements) article says no (though it adds OneNote). [JSON SOT](https://github.com/OfficeDev/Office-Js-Requirement-Sets/blob/master/mapping/requirements_officejs.json) also doesn't note Word as supporting ContentApp.

> **Content add-ins** are surfaces that can be embedded directly into Excel or PowerPoint documents. Content add-ins give users access to interface controls that run code to modify documents or display data from a data source. Use content add-ins when you want to embed functionality directly into the document. See [Content Office Add-ins](../design/content-add-ins.md) to learn more.

Reviewer 2: @alison-mk Also consider updating the Content Add-ins article even in a subsequent PR

## custom function

> In the Office Add-ins documentation, a **custom function** is a user-defined function in Excel. Custom functions in Excel enable developers to add new functions, beyond the typical Excel features, by defining those functions in JavaScript as part of an add-in. Users within Excel can access custom functions just as they would any native function in Excel. See [Create custom functions in Excel](../excel/custom-functions-overview.md) to learn more.

Author: My goal here is to reference "user-defined function" as a general programming concept, such as: https://en.wikipedia.org/wiki/User-defined_function. What if I change the sentence to "a custom function is a type of user-defined function in Excel"?

> In the Office Add-ins documentation, a **custom function** is a user-defined function in Excel. Custom functions in Excel enable developers to add new functions, beyond the typical Excel features, by defining those functions in JavaScript as part of an add-in. Users within Excel can access custom functions just as they would any native function in Excel. See [Create custom functions in Excel](../excel/custom-functions-overview.md) to learn more.

Reviewer 3: What if we emphasize the add-ins model that custom functions have? How about "a **custom function** is a user-defined function that is packaged with an Excel add-in" (or something to that effect). If you don't want to go that route, I think I prefer your original text.

## host

In the Office Add-ins documentation, **host** typically refers to an Office application. The Office applications, or hosts, that support Office Add-ins are Excel, OneNote, Outlook, PowerPoint, Project, Visio, and Word.

See also: [application](#application), [client](#client), [Office application, Office client](#office-application-office-client).

## Office application, Office client

In the Office Add-ins documentation, **Office client** refers to an Office application. The Office applications, or clients, that support Office Add-ins are Excel, OneNote, Outlook, PowerPoint, Project, Visio, and Word.

See also: [application](#application), [client](#client), [host](#host).

## platform

> In the Office Add-ins context, a **platform** usually refers to the operating system running an add-in. Platforms that support Office Add-ins are Windows, Mac, iPad, and web browsers.

Reviewer 2: ```suggestion
In the Office Add-ins context, a **platform** usually refers to the operating system running an add-in. Platforms that support Office Add-ins include Windows, Mac, iPad, and web browsers.
```

## requirement set

[!include[Requirement set note](../includes/office-js-requirement-sets.md)]

## ribbon, ribbon button

A **ribbon** is a command bar that organizes an application's features into a series of tabs or buttons at the top of a window. A **ribbon button** is one of the buttons within this series.

## runtime

> In the Office Add-ins context, a **runtime** is a lifecycle, or the time during which an application is running.

Author: My understanding is that shared runtime ensures ribbon, custom function, and task pane code all run in the same lifecycle. 

@davidchesnut Could you provide feedback on the definition of **runtime** in this Glossary PR? 

Thanks!

## task pane

> Task panes are interface surfaces that typically appear on the right side of the window within Word, PowerPoint, Excel, and Outlook. Task panes give users access to interface controls that run code to modify documents or emails, or display data from a data source. Use task panes when you don't need to embed functionality directly into the document. See [Task panes in Office Add-ins](../design/task-pane-add-ins.md) to learn more.

Author: I will leave this as an explicit list, because my understanding is that Excel, Outlook, PowerPoint, and Word are the only applications that support task panes (and that Project, OneNote, and Visio do not support task panes).

## tutorial

> A **tutorial** is a teaching aid designed to help people learn to use a product or procedure. In the Office Add-ins context, a tutorial guides an add-in developer through the complete add-in development process for a particular application, such as Excel. This involves following 20 or more steps and is a greater time investment than a [quickstart](#quickstart).
> 
> See also: [quickstart](#quickstart).

Reviewer 2: ```suggestion
A **tutorial** is a teaching aid designed to help people learn to use a product or procedure. In the Office Add-ins context, a tutorial guides an add-in developer through the complete add-in development process for a particular application, such as Excel. This involves following 20 or more steps and is a greater time investment than a [quick start](#quick-start).

See also: [quick start](#quick-start).
```

> ## quickstart

Author: Thanks, good catch! I forgot the alphabet.

> ## quickstart

Author: Thanks, good catch! I forgot the alphabet.

A **quickstart** is a a high level description of key skills and knowledge required for the basic operation of a particular program. In the Office Add-ins documentation, a quickstart is an introduction to developing an add-in for a particular application, such as Outlook. A quickstart contains series of steps that an add-in developer can complete in approximately 5 minutes, resulting in a functioning add-in.

See also: [tutorial](#tutorial).

## UI-less custom function

> A **UI-less custom function** is a custom functions add-in that doesn't have a task pane or other user-interface elements.

Author: I see your point, but since custom functions is never hyphenated I want to avoid it. Do you think italicizing "custom functions add-in" would help in this context?

> A **UI-less custom function** is a custom functions add-in that doesn't have a task pane or other user-interface elements.

Reviewer 2: I suggested the hyphen based on how I understood what's said in the [MS style guide](https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=28765) re: noun modifiers

See also: [custom function](#custom-function).

## web add-in

> In the Office Add-ins documentation, **web add-in** is a legacy term for Office Add-ins.

Reviewer 3: Maybe add a line like "It is used in the Microsoft 365 documentation to differentiate between modern Office Add-ins and older COM or VSTO add-ins."

See also: [add-in](#add-in).

## XLL

> An **XLL** add-in is an Excel add-in file with the file extension **.xll**. An XLL file is a type of dynamic link library (DLL) file that can only be opened by Excel. XLL add-in files must be written in C or C++. See [Developing Excel XLLs](/office/client-developer/excel/developing-excel-xlls) to learn more.

Reviewer 3: Consider addressing the outdated nature of XLLs.

Reviewer 3: I like it. Let's tag @davidchesnut to double check.

See also: [custom function](#custom-function).

## Yeoman generator, yo office

> The [Yeoman generator for Office Add-ins](https://github.com/OfficeDev/generator-office) uses the open source [Yeoman](https://github.com/yeoman/yo) tool to generate an Office Add-in via the command line. **Yo office** is the command line argument that runs the Yeoman generator for Office Add-ins. The Office Add-ins quickstarts and tutorials use the Yeoman generator.

Reviewer 3: Since this refers to the command line argument, consider stylizing the casing accordingly.
```suggestion
The [Yeoman generator for Office Add-ins](https://github.com/OfficeDev/generator-office) uses the open source [Yeoman](https://github.com/yeoman/yo) tool to generate an Office Add-in via the command line. **yo office** is the command line argument that runs the Yeoman generator for Office Add-ins. The Office Add-ins quickstarts and tutorials use the Yeoman generator.
```

## add-in

Office Add-ins are web applications that extend Office applications. These web applications add new functionality to the Office application, such as bring in external data, automate processes, or embed interactive objects in Office documents.

Office Add-ins differ from VBA, COM, and VSTO add-ins because they offer cross-platform support (web, Windows, Mac, and iPad) and are based on standard web technologies (HTML, CSS, and JavaScript). The primary programming language of an Office Add-in is JavaScript or TypeScript.

## add-in commands

**Add-in commands** are UI elements, such as buttons and menus, that extend the Office UI for your add-in. When users select an add-in command element, they initiate actions such as running JavaScript code or displaying the add-in in a task pane. Add-in commands let your add-in look and feel like a part of Office, which gives users more confidence in your add-in. See [Add-in commands for Excel, PowerPoint, and Word](../design/add-in-commands.md) and [Add-in commands for Outlook](../outlook/add-in-commands-for-outlook.md) to learn more.

See also: [ribbon, ribbon button](#ribbon-ribbon-button).

## application

> **Application** refers to an Office application. The Office applications that support Office Add-ins are Excel, OneNote, Outlook, PowerPoint, Project, Visio, and Word.

Reviewer 3: I don't think Visio technically supports add-ins. The Visio JS APIs allow for Visio diagrams to be embedded in SharePoint.

Author: My inclination is to simply remove Visio from this list. Do you think that's reasonable?

See also: [client](#client), [host](#host), [Office application, Office client](#office-application-office-client).

## application-specific API

> Application-specific APIs provide strongly-typed objects that interact with objects that are native to a specific Office application. For example, you call the Excel JavaScript APIs for access to worksheets, ranges, tables, charts, and more. Application-specific APIs are currently available for Excel, OneNote, PowerPoint, and Word. See [Application-specific API model](../develop/application-specific-api-model.md) to learn more.

Reviewer 3: Counter to the earlier point, Visio does have an application-specific API. In fact, that's all it has because it can't use the Common APIs (the exception being OfficeExtension APIs, which are technically part of the application-specific model, but that level of detail doesn't need to be here).

Author: I'll add Visio to this list.

See also: [Common API](#common-api).

## client

**Client** typically refers to an Office application. The Office applications, or clients, that support Office Add-ins are Excel, OneNote, Outlook, PowerPoint, Project, Visio, and Word.

See also: [application](#application), [host](#host), [Office application, Office client](#office-application-office-client).

## Common API

Common APIs can be used to access features such as UI, dialogs, and client settings that are common across multiple Office applications. This API model uses [callbacks](https://developer.mozilla.org/docs/Glossary/Callback_function), which allow you to specify only one operation in each request sent to the Office application.

> Common APIs were introduced with Office 2013 and can be used to interact with Office 2013 or later. Some Common APIs, particularly the Common APIs related to Excel, are legacy APIs from the early 2010s. The application-specific APIs are preferred when possible. Other Common APIs, such as the Common APIs related to Outlook, UI, and authentication, are the modern and preferred APIs for these purposes. For details about the Common API object model, which includes APIs for interacting with Outlook, PowerPoint, and Project, see [Common JavaScript API object model](../develop/office-javascript-api-object-model.md).

Reviewer 3: The callout to PowerPoint might be misleading here. WXP are in the same boat: they have application-specific APIs and also functionality from the Common APIs. Maybe it's worth mentioning that WXP have Common API functionality, but most of it has been replaced by the application-specific model?

## Common API

Common APIs are used to access features such as UI, dialogs, and client settings that are common across multiple Office applications. This API model uses [callbacks](https://developer.mozilla.org/docs/Glossary/Callback_function), which allow you to specify only one operation in each request sent to the Office application.

Common APIs were introduced with Office 2013 and are used to interact with Office 2013 or later. Some Common APIs are legacy APIs from the early 2010s. Excel, PowerPoint, and Word all have Common API functionality, but most of this functionality has been replaced or superseded by the application-specific API model. The application-specific APIs are preferred when possible.

Other Common APIs, such as the Common APIs related to Outlook, UI, and authentication, are the modern and preferred APIs for these purposes. For details about the Common API object model, see [Common JavaScript API object model](../develop/office-javascript-api-object-model.md).

See also: [application-specific API](#application-specific-api).

## content add-in

> **Content add-ins** are webviews, or web browser views, that are embedded directly into Excel, PowerPoint, or OneNote documents. Content add-ins give users access to interface controls that run code to modify documents or display data from a data source. Use content add-ins when you want to embed functionality directly into the document. See [Content Office Add-ins](../design/content-add-ins.md) to learn more.

Reviewer 2: ```suggestion
**Content add-ins** are webviews, or web browser views, that are embedded directly into Excel, OneNote, or PowerPoint documents. Content add-ins give users access to interface controls that run code to modify documents or display data from a data source. Use content add-ins when you want to embed functionality directly into the document. See [Content Office Add-ins](../design/content-add-ins.md) to learn more.
```

## ribbon, ribbon button

A **ribbon** is a command bar that organizes an application's features into a series of tabs or buttons at the top of a window. A **ribbon button** is one of the buttons within this series. See [Show or hide the ribbon in Office](https://support.microsoft.com/office/d946b26e-0c8c-402d-a0f7-c6efa296b527#ID0EBBD=Newer_Versions) for more information.

## runtime

See: [JavaScript runtime](#javascript-runtime).

## shared JavaScript runtime, shared runtime

A **shared JavaScript runtime**, or **shared runtime**, enables all code in your add-in, including task pane, add-in commands, and custom functions, to run in the same JavaScript runtime and continue running even when the task pane is closed. See [Configure your Office Add-in to use a shared JavaScript runtime](../develop/configure-your-add-in-to-use-a-shared-runtime.md) and [Tips for using the shared JavaScript runtime in your Office Add-in](https://devblogs.microsoft.com/microsoft365dev/tips-for-using-the-shared-javascript-runtime-in-your-office-add-in%e2%80%af/) to learn more.

See also: [custom functions runtime](#custom-functions-runtime), [JavaScript runtime](#javascript-runtime).

> 

Reviewer: Other terms to consider: SaaS, deploy/centralized deployment, integrated apps, SAML applications, manifest, FRE
